tracer/src/Datadog.Trace.ClrProfiler.Native/rejit_work_offloader.h

```diff
@@ -0,0 +1,51 @@
+#ifndef DD_CLR_PROFILER_REJIT_WORK_OFFLOADER_H_
+#define DD_CLR_PROFILER_REJIT_WORK_OFFLOADER_H_
+
+#include <atomic>
+#include <future>
+#include <mutex>
+#include <shared_mutex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "cor.h"
+#include "corprof.h"
+#include "module_metadata.h"
+
+namespace trace
+{
+
+struct RejitWorkItem
+{
+    bool terminating = false;
```

Comment:
I'm not fully aware of the  C++ threading model, but I'm wondering: should you make it `volatile`
or use `std::atomic<bool>` ?

Reply:
The `RejitWorkItem` structure is used as a payload to `RejitWorkOffloader`, and upon its creation the fields never changes.
I'll mark both `terminating` and `func` as `const` to better communicate their usage. Thanks!

Reply:
Done in f51f557a65cc77300eb9ed96b305542882ece9cb.